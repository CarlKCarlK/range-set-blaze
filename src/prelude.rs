//! This prelude module provides a convenient way to import the most commonly used
//! types, traits, and functions.
//!
//! ```
//! use range_set_blaze::prelude::*;
//! ```
pub use crate::{
    intersection_dyn, intersection_map_dyn, union_dyn, union_map_dyn, CheckSortedDisjoint,
    DynSortedDisjoint, DynSortedDisjointMap, MultiwayRangeMapBlaze, MultiwayRangeSetBlaze,
    MultiwaySortedDisjoint, MultiwaySortedDisjointMap, RangeMapBlaze, RangeSetBlaze,
    SortedDisjoint, SortedDisjointMap, SortedStarts,
};
