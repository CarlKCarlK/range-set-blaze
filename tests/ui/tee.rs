use itertools::Itertools;
#[allow(unused_imports)]
use range_set_blaze::prelude::*;

fn main() {
    let a = [(1, -1), (-3, 4)];
    let (a0, a1) = a.iter().tee();
    let _c = a0.union(a1);
}
