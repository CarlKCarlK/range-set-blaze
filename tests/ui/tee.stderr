error[E0599]: the method `union` exists for struct `Tee<Iter<'_, ({integer}, {integer})>>`, but its trait bounds were not satisfied
 --> tests/ui/tee.rs:7:17
  |
7 |     let _c = a0.union(a1);
  |                 ^^^^^ method cannot be called on `Tee<Iter<'_, ({integer}, {integer})>>` due to unsatisfied trait bounds
  |
 ::: $CARGO/itertools-0.10.5/src/tee.rs
  |
  | pub struct Tee<I>
  | -----------------
  | |
  | doesn't satisfy `<_ as IntoIterator>::Item = &RangeSetBlaze<_>`
  | doesn't satisfy `<_ as IntoIterator>::Item = RangeSetBlaze<_>`
  | doesn't satisfy `_: MultiwayRangeSetBlaze<'_, _>`
  | doesn't satisfy `_: MultiwayRangeSetBlazeRef<_>`
  | doesn't satisfy `_: SortedDisjoint<_>`
  |
 ::: $RUST/core/src/slice/iter.rs
  |
  | pub struct Iter<'a, T: 'a> {
  | -------------------------- doesn't satisfy `_: SortedDisjoint<_>`
  |
  = note: the following trait bounds were not satisfied:
          `<Tee<core::slice::Iter<'_, ({integer}, {integer})>> as IntoIterator>::Item = &RangeSetBlaze<_>`
          which is required by `Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwayRangeSetBlaze<'_, _>`
          `<Tee<core::slice::Iter<'_, ({integer}, {integer})>> as IntoIterator>::Item = RangeSetBlaze<_>`
          which is required by `Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwayRangeSetBlazeRef<_>`
          `core::slice::Iter<'_, ({integer}, {integer})>: range_set_blaze::SortedDisjoint<_>`
          which is required by `Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::SortedDisjoint<_>`
          `<&Tee<core::slice::Iter<'_, ({integer}, {integer})>> as IntoIterator>::Item = &RangeSetBlaze<_>`
          which is required by `&Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwayRangeSetBlaze<'_, _>`
          `&Tee<core::slice::Iter<'_, ({integer}, {integer})>>: IntoIterator`
          which is required by `&Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwayRangeSetBlaze<'_, _>`
          `<&Tee<core::slice::Iter<'_, ({integer}, {integer})>> as IntoIterator>::Item = RangeSetBlaze<_>`
          which is required by `&Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwayRangeSetBlazeRef<_>`
          `&Tee<core::slice::Iter<'_, ({integer}, {integer})>>: IntoIterator`
          which is required by `&Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwayRangeSetBlazeRef<_>`
          `<&Tee<core::slice::Iter<'_, ({integer}, {integer})>> as IntoIterator>::Item = _`
          which is required by `&Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwaySortedDisjoint<_, _>`
          `&Tee<core::slice::Iter<'_, ({integer}, {integer})>>: IntoIterator`
          which is required by `&Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwaySortedDisjoint<_, _>`
          `<&mut Tee<core::slice::Iter<'_, ({integer}, {integer})>> as IntoIterator>::Item = &RangeSetBlaze<_>`
          which is required by `&mut Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwayRangeSetBlaze<'_, _>`
          `<&mut Tee<core::slice::Iter<'_, ({integer}, {integer})>> as IntoIterator>::Item = RangeSetBlaze<_>`
          which is required by `&mut Tee<core::slice::Iter<'_, ({integer}, {integer})>>: range_set_blaze::MultiwayRangeSetBlazeRef<_>`

warning: unused import: `range_set_blaze::prelude`
 --> tests/ui/tee.rs:2:5
  |
2 | use range_set_blaze::prelude::*;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
